packages/engine/src/scene/materials/systems/MaterialLibrarySystem.tsx

```diff
@@ -45,12 +65,97 @@ const reactor = () => {
     )
     const fallbackMaterial = new MeshBasicMaterial({ name: 'Fallback Material', color: 0xff69b4 })
     fallbackMaterial.uuid = MaterialStateComponent.fallbackMaterial
-    createAndAssignMaterial(UndefinedEntity, fallbackMaterial)
+    const fallbackMaterialEntity = createEntity()
+    setComponent(fallbackMaterialEntity, MaterialStateComponent, {
+      material: fallbackMaterial,
+      instances: [UndefinedEntity]
+    })
+    setComponent(fallbackMaterialEntity, UUIDComponent, MaterialStateComponent.fallbackMaterial)
+    setComponent(fallbackMaterialEntity, NameComponent, 'Fallback Material')
   }, [])
 
+  const rendererState = useMutableState(RendererState)
+  useEffect(() => {
```

Comment:
rather than mutating the state here, we should be using `qualityLevel === 0 || forceBasicMaterials === true ` in any effects that depend on `forceMasicMaterials`. When testing locally, the I would turn on basic materials from the debug menu, and (likely due to the few seconds of lag experienced as materials are switched) it would turn back off (because the qualityLevel is changing and triggering the useEffect)

Edit: In fact, it looks like `tieredSettings` is in the performance scaler is doing this too, which is also causing problems with the setting locally. The material changes themselves look fine, so maybe we can address this in a followup.

Reply:
@HexaField Yeah originally my solution was to do just this, but to keep consistency with that preexisting functionality I added this useEffect. Weirdly the tieredSettings config seems to only be properly applied the first time you set quality level to 0 in the settings menu but not when it's 0 on load.